From c833c69607257ecbfcd74eb3299c8aaac531eeb8 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Fri, 19 May 2017 18:24:34 +0300
Subject: [PATCH 0844/5242] MLK-14977: ARM: dts: imx7: Use IMX7D_GPT_3M_CLK
 for GPT1

commit  2e0115c7f7794048b9a9a57b7232408b65faef92 from
https://source.codeaurora.org/external/imx/linux-imx.git

On most imx SOCs GPT1 takes it's clock from the oscillator because
otherwise it might get confusing when bus frequency is decreased.

Right now imx7 is an exception because imx7s.dtsi comes from upstream
rather than a port of imx_4.1.y.

On the imx_4.1.y branch imx7 uses GPT_3M as well, adopt that approach.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7s.dtsi |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx7s.dtsi b/arch/arm/boot/dts/imx7s.dtsi
index d0844b7..4f12a2b 100644
--- a/arch/arm/boot/dts/imx7s.dtsi
+++ b/arch/arm/boot/dts/imx7s.dtsi
@@ -445,8 +445,9 @@
 				reg = <0x302d0000 0x10000>;
 				interrupts = <GIC_SPI 55 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&clks IMX7D_GPT1_ROOT_CLK>,
-					 <&clks IMX7D_GPT1_ROOT_CLK>;
-				clock-names = "ipg", "per";
+					<&clks IMX7D_GPT1_ROOT_CLK>,
+					<&clks IMX7D_GPT_3M_CLK>;
+				clock-names = "ipg", "per", "osc_per";
 			};
 
 			gpt2: gpt@302e0000 {
-- 
1.7.9.5

